From cb727df508fe733a7b0937634674ffe8f4fcba06 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 12 Dec 2021 10:21:02 +0100
Subject: [PATCH] intl dep: forward static to iconv if needed

In 1fb6c939473ca7cdab2 the intl dep gained support for static linking
which also forwarded this property to the iconv sub dependency.

The refactoring in 214d03568f75 lost this change, which results in
iconv getting linked dynamically again.

Forward static again to fix this.
---
 mesonbuild/dependencies/misc.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/dependencies/misc.py b/mesonbuild/dependencies/misc.py
index e490e7dee59..470305e91a7 100644
--- a/mesonbuild/dependencies/misc.py
+++ b/mesonbuild/dependencies/misc.py
@@ -487,7 +487,7 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
             self.is_found = True
 
             if self.static:
-                if not self._add_sub_dependency(iconv_factory(env, self.for_machine, {})):
+                if not self._add_sub_dependency(iconv_factory(env, self.for_machine, {'static': True})):
                     self.is_found = False
                     return
 
